packages/lib/src/__tests__/output-builder.test.ts

```diff
@@ -1,22 +1,138 @@
+import * as output from '../output'
+import { buildOutputFormatter } from '../output-builder'
+import * as ioUtil from '../io-util'
+import { SmartThingsCommandInterface } from '../smartthings-command'
+
+import { buildMockCommand } from './input-builder.test'
+import { SimpleType } from './io-util.test'
+
 
 describe('buildOutputFormatter', () => {
+	let command: SmartThingsCommandInterface
+	let jsonFormatterSpy: jest.SpyInstance<output.OutputFormatter<unknown>, [indent: number]>
+	let yamlFormatterSpy: jest.SpyInstance<output.OutputFormatter<unknown>, [indent: number]>
+	let formatFromFilenameSpy: jest.SpyInstance<ioUtil.IOFormat, [filename: string]>
+
+	beforeEach(() => {
+		command = buildMockCommand()
+		jsonFormatterSpy = jest.spyOn(output, 'jsonFormatter')
+		yamlFormatterSpy = jest.spyOn(output, 'yamlFormatter')
+		formatFromFilenameSpy = jest.spyOn(ioUtil, 'formatFromFilename')
+	})
+
+	afterEach(() => {
+		jest.clearAllMocks()
+	})
+
 	it('uses json when specified', () => {
-		// buildOutputFormatter(command, )
+		command.flags.json = true
+
+		buildOutputFormatter<SimpleType>(command)
+		expect(jsonFormatterSpy).toHaveBeenCalledTimes(1)
+
+		jsonFormatterSpy.mockReset()
+		command.flags.input = 'fn.yaml'
+		buildOutputFormatter<SimpleType>(command)
+		expect(jsonFormatterSpy).toHaveBeenCalledTimes(1)
+	})
+
+	it('uses yaml when specified', () => {
+		command.flags.yaml = true
+
+		buildOutputFormatter<SimpleType>(command)
+
+		expect(yamlFormatterSpy).toHaveBeenCalledTimes(1)
+
+		yamlFormatterSpy.mockReset()
+		command.flags.input = 'fn.json'
+		buildOutputFormatter<SimpleType>(command)
+		expect(yamlFormatterSpy).toHaveBeenCalledTimes(1)
+	})
+
+	it('gets format from filename with input file when not specified', () => {
+		formatFromFilenameSpy.mockReturnValue(ioUtil.IOFormat.JSON)
+		command.flags.output = 'fn.json'
+		buildOutputFormatter<SimpleType>(command)
+		expect(formatFromFilenameSpy).toHaveBeenCalledTimes(1)
+		expect(formatFromFilenameSpy).toHaveBeenCalledWith('fn.json')
+		expect(jsonFormatterSpy).toHaveBeenCalledTimes(1)
+		expect(yamlFormatterSpy).toHaveBeenCalledTimes(0)
+
+		jest.resetAllMocks()
```

Comment:
It would be valuable in some of these cases to split into separate tests. For example, a JSON case and YAML case. It provides more context with failing tests (ex. both are failing or one is) and narrows time to find issues.

Reply:
I actually just deleted the second one instead...it was a duplicate rather than a separate test. The tests for the different file types are already taken care of (correctly) by unit tests for `formatFromFilename` itself.